src/synapse/azext_synapse/commands.py

```diff
@@ -0,0 +1,96 @@
+        client_factory=cf_synapse_spark_batch
+    )
+
+    synapse_spark_session_sdk = CliCommandType(
+        operations_tmpl='azext_synapse.vendored_sdks.azure_synapse.operations#SparkSessionOperations.{}',
+        client_factory=cf_synapse_spark_session
+    )
+
+    # Management Plane Commands
+    with self.command_group('synapse workspace', synapse_workspace_sdk,
+                            client_factory=cf_synapse_client_workspace_factory) as g:
+        g.show_command('show', 'get')
+        g.custom_command('list', 'list_workspaces')
+        g.custom_command('create', 'create_workspace', supports_no_wait=True)
+        g.custom_command('update', 'update_workspace', supports_no_wait=True)
+        g.command('delete', 'delete', confirmation=True, supports_no_wait=True)
+        g.wait_command('wait')
+
+    with self.command_group('synapse spark pool', synapse_bigdatapool_sdk,
+                            client_factory=cf_synapse_client_bigdatapool_factory) as g:
+        g.show_command('show', 'get')
+        g.command('list', 'list_by_workspace')
+        g.custom_command('create', 'create_spark_pool', supports_no_wait=True)
+        g.custom_command('update', 'update_spark_pool', supports_no_wait=True)
+        g.command('delete', 'delete', confirmation=True, supports_no_wait=True)
+        g.wait_command('wait')
+
+    with self.command_group('synapse sql pool', synapse_sqlpool_sdk,
+                            client_factory=cf_synapse_client_sqlpool_factory) as g:
+        g.show_command('show', 'get')
+        g.command('list', 'list_by_workspace')
+        g.custom_command('create', 'create_sql_pool', supports_no_wait=True)
+        g.command('delete', 'delete', confirmation=True, supports_no_wait=True)
+        g.command('pause', 'pause')
+        g.command('resume', 'resume')
+        g.wait_command('wait')
+
+    # Data Plane Commands
+    # Spark batch opertions
+    with self.command_group('synapse spark batch', synapse_spark_batch_sdk, client_factory=cf_synapse_spark_batch) as g:
+        g.custom_command('create', 'create_spark_batch_job', supports_no_wait=True)
+        g.custom_command('list', 'list_spark_batch_jobs')
+        g.custom_command('show', 'get_spark_batch_job')
+        g.custom_command('cancel', 'cancel_spark_batch_job', confirmation=True, supports_no_wait=True)
+        g.wait_command('wait')
+
+    # Spark session operations
+    with self.command_group('synapse spark session', synapse_spark_session_sdk,
+                            client_factory=cf_synapse_spark_session) as g:
+        g.custom_command('create', 'create_spark_session_job', supports_no_wait=True)
+        g.custom_command('list', 'list_spark_session_jobs')
+        g.custom_command('show', 'get_spark_session_job')
+        g.custom_command('cancel', 'cancel_spark_session_job', confirmation=True, supports_no_wait=True)
+        g.custom_command('reset-timeout', 'reset_timeout')
+        g.wait_command('wait')

```

Comment:
Could you please verify `wait` command takes effect here?

Reply:
removed

---
In reply to: [386410157](https://github.com/Azure/azure-cli-extensions/pull/1339#discussion_r386410157) [](ancestors = 386410157)